### PR TITLE
Fix modify parsing bugs, add line separator for "mi"

### LINKED ISSUE #265: Bug: Modify income has no visual line separators
Image

### LINKED ISSUE #266: Bug: Modify income doesn't reject when no fields specified
Modify expense correctly rejects when no fields specified, but modify income does not. It simply re-prints the original income.

Image

### LINKED ISSUE #267: Bug: Both modify expense and modify income don't reject stray text properly
This even breaks modify expense's detection for no fields

me:

Image

mi:

Image

### LINKED ISSUE #270: Cannot Clear Description via `modify` Commands
# Cannot Clear Description via `modify` Commands

## Summary
Users cannot remove an existing description from an expense or income using the `modify` commands. Supplying `des/` with no value leaves the old description untouched instead of clearing it.

## Steps to Reproduce
1. `add-expense a/1.00 c/food d/2024-10-01 des/tasty`
2. `modify-expense 1 des/`
3. `list-expense`

## Expected Result
The expense prints with no description after step 2.

## Actual Result
The expense still shows the original description (`tasty`).

## Impact
Descriptions become sticky: once a note is added to an entry, it cannot be removed via the CLI. This affects both expenses and incomes, since `modify-income` shares the same parse logic.

### PR DESCRIPTION
**Pull request by sanjay-shiva-kumar**
Fixes #265 
fixes #266 
fixes #267 
fixes #270

### CODE REVIEW

File: src/main/java/seedu/fintrack/FinanceManager.java [hunk 1]
```diff
@@ -492,7 +492,7 @@ public double calculatePercentage(double amount, double totalAmount) {
         return percent;
     }
 
-    public Map<ExpenseCategory, Double>getExpensePercentageByCategory(
+    public Map<ExpenseCategory, Double> getExpensePercentageByCategory(
```
Comment (duckyfuz) [inline src/main/java/seedu/fintrack/FinanceManager.java]: another great formatting catch, thanks!

File: src/test/java/seedu/fintrack/ParserTest.java [hunk 2]
```diff
@@ -916,17 +978,86 @@ public void parseModifyIncomeWithDefaults_multipleFields_overridesOnlySpecified(
...
-    public void parseModifyIncomeWithDefaults_noFields_keepsAllFields() {
-        Income oldIncome = new Income(777.77, IncomeCategory.INVESTMENT,
-                LocalDate.of(2025, 8, 8), "dividends");
+    public void parseModifyIncomeWithDefaults_noFields_throws() {
+        Income oldIncome = new Income(
+                777.77,
+                IncomeCategory.INVESTMENT,
+                LocalDate.of(2025, 8, 8),
+                "dividends"
+        );
```
Comment (duckyfuz) [inline src/test/java/seedu/fintrack/ParserTest.java]: good formatting done here

File: src/test/java/seedu/fintrack/ParserTest.java [hunk 3]
```diff
@@ -916,17 +978,86 @@ public void parseModifyIncomeWithDefaults_multipleFields_overridesOnlySpecified(
...
+    public void parseModifyIncomeWithDefaults_blankAmount_throws() {
+        Income oldIncome = new Income(900, IncomeCategory.GIFT,
+                LocalDate.of(2025, 3, 3), "bonus");
+
+        IllegalArgumentException ex = assertThrows(
+                IllegalArgumentException.class,
+                () -> Parser.parseModifyIncomeWithDefaults(Ui.MODIFY_INCOME_COMMAND + " 1 a/", oldIncome)
+        );
+        assertEquals("Amount must follow a/.", ex.getMessage());
+    }
```
Comment (duckyfuz) [inline src/test/java/seedu/fintrack/ParserTest.java]: Indeed this is a great edge case to check for
